src/mscorlib/shared/System/IO/FileStream.Win32.cs

```diff
@@ -3,14 +3,27 @@
 // See the LICENSE file in the project root for more information.
 
 using Microsoft.Win32.SafeHandles;
-using System.Runtime.InteropServices;
 
 namespace System.IO
 {
     public partial class FileStream : Stream
     {
+        private static bool UseFromApp() => AppDomain.IsAppXModel();
+
         private SafeFileHandle OpenHandle(FileMode mode, FileShare share, FileOptions options)
         {
+#if PROJECTN || CORERT
+            return CreateFile2OpenHandle(mode, share, options);
+#else
+            // CreateFile2 isn't available on Windows 7
+            return Environment.IsWindows8OrAbove
```

Comment:
I was trying to get most of our code on the same API, but I'm fine with reducing this to one check.